lib/greet.js

```diff
@@ -0,0 +1,11 @@
+'use strict';
+
+module.exports = exports = {};
+
+exports.greet = function(name) {
+  console.log(`hello ${name}`);
+  if (!arguments.length) throw new Error('name not provided');
+  return `hello ${name}!`;
+};
+
+console.log('hello', process.argv[2]);
```

Comment:
Since you are just exporting a single function in this case, module.exports = function ( ) would be more appropriate. You want to use module.exports = exports = { } when you are exporting multiple methods. For instance, if you had another method attached to the exports object in this file, then the way you are using module exports on line 3 would be just fine :)